src/backend/joanie/core/models/courses.py

```diff
@@ -327,8 +336,23 @@ def __str__(self):
         )
 
     def clean(self):
-        """Normalize the resource_link url."""
+        """
+        Normalize the resource_link url and prevent data integrity error when course is
+        updated.
+        """
         self.resource_link = url_normalize(self.resource_link)
+
+        # If the course run is updating and the course field has changed ...
+        if self.pk and self.get_field_diff("course_id") is not None:
+            # ... Check the course run instance does not rely on product/order relations
+            if self.product_relations.count() > 0 or self.order_relations.count() > 0:
+                raise ValidationError(
+                    _(
+                        "This course run relies on a product relation. "
+                        "So you cannot modify its course."
```

Comment:
maybe it's possible to add a constraint on the `course_run` table to ensure that `course_run.course` is equal to the value for the `course` field on all `course_run.product_relations` for this course run.

Reply:
Unfortunately, use a many to many field in a constraint is forbidden.